course_content/notebooks/dask_intro.ipynb

```diff
@@ -1241,6 +1241,314 @@
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**3.**  \n",
+    "Have the data been loaded ?  \n",
+    "Are the mean cubes still lazy ?  \n",
+    "What about the original u/v cubes ?  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Iris in-built lazy operations : cube arithmetic\n",
+    "\n",
+    "We can easily calculate windspeeds as a vector magnitude.  \n",
+    "This can be made from the U and V cubes using the Iris \"cube arithmetic\" operators.  \n",
+    "The result is still lazy -- i.e. data has not yet been loaded from disk."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cube_windspeed = (u_cube * u_cube + v_cube * v_cube) ** 0.5\n",
+    "cube_windspeed.rename('wind_speed')\n",
+    "\n",
+    "print 'Windspeed cube:'\n",
+    "print cube_windspeed\n",
+    "print\n",
+    "print 'Windspeed cube : lazy = ', cube_windspeed.has_lazy_data()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Custom lazy calculations\n",
+    "\n",
+    "Suppose though that we want to calculate wind _**directions**_.  \n",
+    "\n",
+    "This is somewhat harder, as Iris doesn't provide a function for the calculation.  \n",
+    "We can, however, create a deferred calculation with Dask and embed that in a cube.  \n",
+    "\n",
+    "For example, like this:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import dask\n",
+    "import dask.array as da\n",
+    "\n",
+    "# Calculate arctan(u, v)\n",
+    "wind_dirs_array = da.arctan2(u_cube.lazy_data(), v_cube.lazy_data())\n",
+    "# Convert to degrees\n",
+    "wind_dirs_array = da.rad2deg(wind_dirs_array)\n",
+    "\n",
+    "# Make a suitable cube + put this into it.\n",
+    "wind_dirs_cube = u_cube.copy(data=wind_dirs_array)\n",
+    "# Fix up the metadata.\n",
+    "wind_dirs_cube.units = 'degrees'\n",
+    "wind_dirs_cube.rename('Wind direction')\n",
+    "\n",
+    "print 'Wind directions cube:',\n",
+    "print wind_dirs_cube\n",
+    "print\n",
+    "print 'Some sample data points : ', wind_dirs_cube[:2, :3].data\n",
+    "print\n",
+    "print 'Wind directions cube : lazy = ', wind_dirs_cube.has_lazy_data()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Exercise : working with a custom lazy calculation\n",
+    "\n",
+    "If you print wind_dirs_cube.data, the cube is no longer lazy\n",
+    "\n",
+    "**1.** Show this happening."
```

Comment:
**1.** Print the cube to demonstrate this.

Reply:
> 1. Print the cube to demonstrate this.

I think that is too spoon-feedy + pointless !
If you just tell people what to type then I think no learning typically occurs ! Instead I intended to ask them to demonstrate something specific, which they need to work out how.
We could instead say "Show that this happens" ??